File: java/sort/InsertionSort.java
class InsertionSort
{
  public static void insertionSort(int a[])
  {
    for(int i=1;i<a.length;i++)
    {
      int j=i;
      while(j>0 && a[j]<a[j-1])
      {
        int t=a[j];
        a[j]=a[j-1];
        a[j-1]=t;
        j--;
      }
    }
  }
}

File: python/sort/insertionsort.py
def INSERTION_SORT(A):
    n = len(A)
    for i in range(1, n):
        x = A[i]
        j = i-1
        while j >= 0 and x < A[j]:
            A[j+1] = A[j]
            j -= 1
        A[j+1] = x
    return A